File: 15/a.go
package main

import (
	"fmt"
	"os"
	"strings"
)

func main() {
	file, _ := os.ReadFile("input.txt")

	res := 0

	for _, str := range strings.Split(string(file), ",") {
		acc := 0
		for _, char := range str {
			acc += int(char)
			acc *= 17
			acc %= 256
		}
		res += acc
	}

	fmt.Println(res)
}

File: 15/a.ts
// const input = await Bun.file("test.txt").text();
const input = await Bun.file("input.txt").text();

const res = input
  .split(",")
  .map((line) => {
    let acc = 0;
    for (const char of line) {
      acc += char.charCodeAt(0);
      acc *= 17;
      acc %= 256;
    }
    return acc;
  })
  .reduce((acc, cur) => acc + cur, 0);

console.log(res);
